From 5adbbc3c4c27e438614487819352158b53b9e5f6 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 5 May 2020 19:45:41 +0100
Subject: [PATCH 214/552] video: bcm2708_fb: Disable FB if no displays found

If the firmware hasn't detected a display, the driver would assume
one display was available, but because it had failed to retrieve the
display size it would try to allocate a zero-sized buffer.

Avoid the allocation failure by bailing out early if no display is
found.

See: https://github.com/raspberrypi/linux/issues/3598

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/video/fbdev/bcm2708_fb.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/video/fbdev/bcm2708_fb.c b/drivers/video/fbdev/bcm2708_fb.c
index 17e9260655e4..dac5c021ce52 100644
--- a/drivers/video/fbdev/bcm2708_fb.c
+++ b/drivers/video/fbdev/bcm2708_fb.c
@@ -1092,10 +1092,9 @@ static int bcm2708_fb_probe(struct platform_device *dev)
 	 * set one display
 	 */
 	if (ret || num_displays == 0) {
-		num_displays = 1;
 		dev_err(&dev->dev,
-			"Unable to determine number of FB's. Assuming 1\n");
-		ret = 0;
+			"Unable to determine number of FBs. Disabling driver.\n");
+		return -ENOENT;
 	} else {
 		fbdev->firmware_supports_multifb = 1;
 	}
-- 
2.33.0

